src/compile/layer.ts

```diff
@@ -104,15 +104,43 @@ export class LayerModel extends Model {
     this.component.layout = parseLayerLayout(this);
   }
 
-  public parseScale() {
+  /**
+   * Convert the domain to an array of data refs. Also, thorw away sorting information
+   * since we always sort the domain when we union two domains.
```

Comment:
>If all data refs have identical sort objects, here you drop all of them and use true instead, which is wrong.

Actually if I understand correctly this case isn't supported in Vega anyway (if there are more than 1 unique fields)?